key-manager/src/lib.rs

```diff
@@ -96,22 +162,65 @@ pub struct KeyManager<S: SecretRetriever> {
     /// reconfigurations before a node is no longer part of the trust quorum.
     /// This should most likely be a small number like `3`.
     prks: BTreeMap<u64, Hkdf<Sha3_256>>,
+
+    // Receives requests from a `StorageKeyRequester`, which is expected to run
+    // in the `StorageWorker` task.
+    storage_rx: mpsc::Receiver<StorageKeyRequest>,
 }
 
 impl<S: SecretRetriever> KeyManager<S> {
-    pub fn new(secret_retriever: S) -> KeyManager<S> {
-        KeyManager { secret_retriever, prks: BTreeMap::new() }
+    pub fn new(secret_retriever: S) -> (KeyManager<S>, StorageKeyRequester) {
+        // There are up to 10 U.2 drives per sleds, but only one request should
+        // come in at a time from a single worker. We leave a small buffer for
+        // the possibility of asynchronous requests without responses that we
+        // may want to send in series.
+        let (tx, rx) = mpsc::channel(10);
+
+        let storage_key_requester = StorageKeyRequester { tx };
+        let key_manager = KeyManager {
+            secret_retriever,
+            prks: BTreeMap::new(),
+            storage_rx: rx,
+        };
+
+        (key_manager, storage_key_requester)
+    }
+
+    /// Run the main receive loop of the `KeyManager`
+    ///
+    /// This should be spawned into a tokio task
+    pub async fn run(&mut self) {
+        loop {
+            tokio::select! {
```

Comment:
This `tokio::select` is here because we will have some other types of key requesters other than `StorageKeyRequester` in the future. We also may want a sled-agent requester that is allowed to clear keys rather than use a timer. Having the select doesn't really hurt if we don't use it, but I'm open to removing it if its confusing and we don't add anything else to select on in this PR.